pkg/operator/operator.go

```diff
@@ -213,3 +219,41 @@ func getVnetGUID(cfg *auth.Config, subnetID string) (string, error) {
 	}
 	return *vnet.Properties.ResourceGUID, nil
 }
+
+// WaitForCRDs waits for the required CRDs to be available with a timeout
+func WaitForCRDs(ctx context.Context, client *discovery.DiscoveryClient, timeout time.Duration, log logr.Logger) error {
+	var requiredCRDs = []schema.GroupVersionResource{
+		{Group: "karpenter.sh", Version: "v1", Resource: "nodepools"},
+		{Group: "karpenter.sh", Version: "v1", Resource: "nodeclaims"},
+		{Group: "karpenter.azure.com", Version: "v1alpha2", Resource: "aksnodeclasses"},
+	}
+
+	log.WithValues("timeout", timeout).Info("waiting for required CRDs to be available")
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	for _, crd := range requiredCRDs {
+		crdName := fmt.Sprintf("%s.%s/%s", crd.Resource, crd.GroupVersion().Group, crd.Version)
+		if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (bool, error) {
+			_, err := client.RESTClient().Get().AbsPath("/apis", crd.Group, crd.Version, crd.Resource).Do(ctx).Raw()
```

Comment:
I think this deserves a comment for why you're avoiding the full `/apis` and using the RESTClient. (I assume to avoid the full /apis list which can be really expensive)

In fact you could probably use `ServerResourcesForGroupVersion` which would still avoid the full /apis query without needing to resort to the raw client. I don't know that this is way clearer though (it might even be more code?) so I'm fine not using it, but think a comment should be here.

Reply:
There was nothing built-in that works nicely, but it looks like RESTMapper from controller-runtime client does what we are looking for, take a look